docs: Integrate MetalK8s installation guide from Zenko

NicolasT opened the pull request:
Integrating the work by @wabernatScality.

See: #308 
See: #285

bert-e commented:
# Hello NicolasT,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

NicolasT commented:
@bert-e wait

NicolasT commented:
`development/1.0` needs to be merged into this one after #324 landed.

Zempashi reviewed (dismissed):
@wabernatScality should have a look on 5435540b219b749e85f3f9e792bebb0bd656bc2f

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,25 +238,34 @@ Similarly, we can list all deployed Helm_ applications::
+   | `Kibana`_               | A search console for logs indexed in Elasticsearch      | http://localhost:8001/api/v1/namespaces/kube-ops/services/http:kibana:/proxy/                   |
+   +-------------------------+---------------------------------------------------------+-------------------------------------------------------------------------------------------------+
+
+See :doc:`../../architecture/cluster-services` for more information about these
```
This seems like a wrong path now? Why is this change required? I'm surprised this even builds.

wabernat replied:
fixed

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
-need to have at least one disk available to provision storage volumes.
-
-.. todo:: Give some sizing examples
+This guide describes how to set up a MetalK8s test cluster. It offers general
```
Could we keep the `MetalK8s` link?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
-configuration.
+General Cluster Requirements
+----------------------------
+Setting up a MetalK8s_ test cluster quickly requires three machines
```
And remove it here :)

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+General Cluster Requirements
+----------------------------
+Setting up a MetalK8s_ test cluster quickly requires three machines
+running CentOS_ 7.4 (these can be VMs) to which you have SSH access. Each
```
7.4 or higher, actually. May want to specify this.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+Sizing
+^^^^^^
+
+Each node must satisfy the following sizing requirements.
```
Would it make sense to put this in a table instead of 3 lists each with the same 3 items?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+Proxies
+^^^^^^^
+For nodes operating behind a proxy, add the following lines to your local
+machine’s /etc/environment file:
```
Apply the `file` role:
```
:file:`/etc/environment`
```

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+
+Proxies
+^^^^^^^
+For nodes operating behind a proxy, add the following lines to your local
```
`local environment file` is confusing! These lines shouldn't be added to your 'local' machine (e.g. Linux running on your laptop), but instead to all cluster servers.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+
+.. code-block:: shell
+
+ http_proxy=http://user;pass@<my-ip>:<my-port>
```
- Why `user` and `pass` but `my-ip` and `my-port`?
- What is `my-ip` and `my-port`? That'd be the HTTP proxy IP and port.
- Sphinx has a `samp` role (http://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-samp) which allows to specify which parts need to be replaced by the user with some custom value. I wonder whether we could apply something like that here...

wabernat replied:
changed to: 
```
http_proxy=http://user;pass@<HTTP proxy IP address>:<port>
https_proxy=http://user;pass@<HTTPS proxy IP address>:<port>
no_proxy=localhost,127.0.0.1,10.*
```

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+
+ http_proxy=http://user;pass@<my-ip>:<my-port>
+ https_proxy=http://user;pass@<my-ip>:<my-port>
+ no_proxy=localhost,127.0.0.1,10.*
```
I believe `10.233.*` should be sufficient, isn't it @xaf-scality ?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+ https_proxy=http://user;pass@<my-ip>:<my-port>
+ no_proxy=localhost,127.0.0.1,10.*
+
+Get the MetalK8s Git Repo
```
I'm still not sure we need to document this. If we need to, this is not the way to go: when following this procedure, one would always end up with the 'latest', non-released codebase (e.g. `development/1.1` right now), which is not necessarily stable. I'd suggest to

- Document where/how to retrieve a released version of MetalK8s (tarball), and use it
- Add a note saying one can also retrieve the sources from Git if one wants to run development versions

wabernat replied:
Where does one find the tarball?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+
+.. code-block:: shell
+
+ $ sftp centos@10.0.0.1
```
I don't think this is the 'right' way to do things. Also see comment above.

wabernat replied:
Please tell me the right way to do this.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+-------------------
+Each server must be configured in an inventory that identifies the servers to
+the Ansible_-based deployment system, as well as their basic configuration,
+including masters, etcds, and nodes.
```
`etcds` has no meaning in this context. May need a section long before this one explaining those 3 roles (they're referred to before as well).

wabernat replied:
Removed reference to etcds.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
+including masters, etcds, and nodes.
+
+The inventory is a directory containing a hosts file, which lists all hosts
+in the cluster, and kube-node.yml, a configuration file.
```
No. `kube-node.yml` doesn't go in the inventory (root) directory. This is confusing. It goes in a subdirectory.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,109 @@
-listing all the hosts, aptly called :file:`hosts`:
+To create an inventory:
+
+1. Log in to the machine to which you cloned or put the git repo, as described
```
Why wouldn't one run the deployment from a local machine?

wabernat replied:
Because there isn't a git repo on it.

wabernat replied:
In other words, my first walk-through of these steps was from my Windows machine, with all its built-in eccentricities. I ended up installing MetalK8s from one of the three nodes. It worked, and introduced a few problems that I have noted and you have questioned. If you want to lay down the law and say that the only way to go about this is to use a local machine to do the install on a remote cluster, that's okay. In any event, I've reworded most of the offending passages to be a little more open-ended.

NicolasT replied:
From a Windows machine, indeed, it may be a bit difficult right now. Current wording is generic enough I think.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -56,100 +132,40 @@ listing all the hosts, aptly called :file:`hosts`:
-
-Given an old configuration looking like this
+   Change the host names, IP addresses, and user names to conform to your
+   infrastructure. For example, if your servers are named “server1”, “server2”,
```
There must be a more fitting role for these examples.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -56,100 +132,40 @@ listing all the hosts, aptly called :file:`hosts`:
 
-.. code-block:: yaml
+4. Create a :file:`group_vars` subdirectory in the directory you created in
+   step 2 (the same directory as the :file:`hosts` file) and change to it.
```
> the same directory as the hosts file

A directory can not be the same as a file.

> the directory containing the hosts file

or something alike?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -56,100 +132,40 @@ listing all the hosts, aptly called :file:`hosts`:
 
-- Disable deployment of 'default' volumes:
+5. Create a file, :file:`kube-node.yml`, in the :file:`group_vars` subdirectory
+   of the inventory. This file declares how to set up storage (in the default
```
You just went into the `group_vars` directory, so this is superfluous or confusing.

wabernat replied:
I would like to maintain this. It keeps people from getting clever ideas about where to put the yaml file. I've reworded it to give better continuity.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -56,100 +132,40 @@ listing all the hosts, aptly called :file:`hosts`:
 
-- Disable deployment of 'default' volumes:
+5. Create a file, :file:`kube-node.yml`, in the :file:`group_vars` subdirectory
+   of the inventory. This file declares how to set up storage (in the default
```
No, `kube-node.yml` doesn't declare how to set up storage. It's a `group_vars` file for the `kube-node` group, nothing more. We will declare storage configuration for members of the `kube-node` group in it though.

wabernat replied:
> In the :file:`group_vars` subdirectory, create a :file:`kube-node.yml` file. 
   This file declares how to set up hosts in the kube-node group; that is, 
   hosts on which pods shall be scheduled:

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -56,100 +132,40 @@ listing all the hosts, aptly called :file:`hosts`:
-          size: 5G
-        lv08:
-          size: 5G
+.. note:: If you are upgrading from an early (pre-0.2.0) MetalK8s release, you must enter additional settings. See “Upgrading from MetalK8s < 0.2.0” (upgrade_from_pre-0.2.0.rst, also in this directory)
```
This should be removed. Also, better use links, don't say

> See ... (filename.rst, also in this directory)

Remember the user is not reading ReST files and isn't browsing directories!

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +175,54 @@ first run)::
+Run the following command to deploy the cluster::
 
-Grab a coffee and wait for deployment to end.
+   (metal-k8s) $ ansible-playbook -i inventory/quickstart-cluster -b playbooks/deploy.yml
```
I know this wasn't in the original doc, but would better be `-i inventory/quickstart-cluster/hosts`.

wabernat replied:
Currently reads:

(metal-k8s) $ ansible-playbook -i inventory/quickstart-cluster/hosts 

Period at end or not?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +175,54 @@ first run)::
-export this location in the shell such that the :program:`kubectl` and
-:program:`helm` tools know how to contact the cluster *kube-master* nodes, and
-authenticate properly::
+Deployment takes about a half hour.
```
This heavily depends on number of nodes, server sizing, network,... Can't this be left out?

wabernat replied:
For a simple test deployment such as the present three-node cluster, this takes
about a half hour. Actual deployment time will vary based on the size of the
cluster and hardware and network performance.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +175,54 @@ first run)::
+-------------------
+Deployment creates a file containing credentials to access the cluster
+(:file:`inventory/quickstart-cluster/artifacts/admin.conf`). Remaining in the
+virtual environment shell, export this locationto give :program:`kubectl` and
```
Missing space, `location to`.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +175,54 @@ first run)::
+Deployment creates a file containing credentials to access the cluster
+(:file:`inventory/quickstart-cluster/artifacts/admin.conf`). Remaining in the
+virtual environment shell, export this locationto give :program:`kubectl` and
+:program:`helm` the correct paths and credentials to contact the cluster’s
```
Which paths?

wabernat replied:
fixed

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +175,54 @@ first run)::
 
-Now, assuming port *6443* on the first *kube-master* node is reachable from
-your system, we can e.g. list the nodes::
+If your system can reach port 6443 on the first kube-master node, you can
```
We need to use some role for group names and stick to it. Currently it's inconsistent.

wabernat replied:
I don't understand the inconsistency. Does the following work? 

"If your system can reach port 6443 on a master node, you can..."

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +175,54 @@ first run)::
-    kube-ops       es-data-cd5446fff-pkmhn                                1/1       Running     0          2m
-    kube-ops       es-data-cd5446fff-zzd2h                                1/1       Running     0          2m
-    kube-ops       es-exporter-elasticsearch-exporter-7df5bcf58b-k9fdd    1/1       Running     3          1m
+    NAMESPACE     NAME                                                 READY  STATUS    RESTARTS  AGE
```
Not sure why there are whitespace changes here, these should be reverted.

wabernat replied:
This is simply to pull the lines in as tight as possible on the screen. Readers are not going to count white spaces, nor is there an interpreter that's going to choke.

NicolasT replied:
The original was copy-pasted from a terminal. If we ever want to update the command output, that would mess things up again :) anyway, fine.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,25 +238,34 @@ Similarly, we can list all deployed Helm_ applications::
+Services to operate and monitor your MetalK8s cluster are provided. To
+access these dashboards:
+
+1. Copy the credentials in
```
If you must deploy from the remote machine in the first place instead of using your local machine as a controller, chances are high you can't reach the K8s master node(s), so this won't work anyway.

If you deployed from the local machine, there's no need to copy whatever files around.

wabernat replied:
This is what I did on the first test cluster I deployed, so that's why I described it thus here. I'll change it to be more general.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,25 +238,34 @@ Similarly, we can list all deployed Helm_ applications::
+   :file:`inventory/quickstart-cluster/artifacts/admin.conf` to your local
+   machine. Export this path locally with::
+
+       $ export KUBECONFIG=`pwd`/inventory/quickstart-cluster/artifacts/admin.conf
```
`pwd` depends on where you downloaded the file etc.

wabernat replied:
changed

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,25 +238,34 @@ Similarly, we can list all deployed Helm_ applications::
+
+       $ export KUBECONFIG=`pwd`/inventory/quickstart-cluster/artifacts/admin.conf
+
+2. On your cluster, open port 6443 for remote access to cluster services.
```
Port on/to which machine?

wabernat replied:
"Open port 6443 on at least one node in your cluster for remote access to cluster services."

wabernat reviewed (dismissed):
Made most of the requested changes. Fine as is, but if you want to push back, just override/overwrite.

NicolasT commented:
@wabernatScality I'm still on this.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,99 @@
++-----------------+--------+--------+-------+------------------+
+| RAM             | 4 GB   | 8 GB   | 8 GB  |                  |
++-----------------+--------+--------+-------+------------------+
+| System Storage  | 16 GB  | 128 GB | 64 GB | 20 GB            |
```
This is confusing. `etcd`, `master` and `node` are 'kinds' of servers. `Root File System` is not. What does this cell refer to?

wabernat replied:
You asked me to put a bunch of bullet points into a table. This is what that looks like.

wabernat replied:
kicked root file system out of the table and into a separate requirement.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,99 @@
+
+.. code-block:: shell
+
+ $ git clone https://github.com/scality/metal-k8s
```
This will get the latest development build. I'll work on a replacement for this paragraph.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,99 @@
+the Ansible_-based deployment system, as well as their basic configuration,
+including masters and nodes.
+
+The inventory is a directory that contains a hosts file, which lists all hosts
```
```
:file:`hosts`
```

wabernat replied:
fixed

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,99 @@
+including masters and nodes.
+
+The inventory is a directory that contains a hosts file, which lists all hosts
+in the cluster, and a subdirectory (group_vars) that contains kube-node.yml, a
```
```
:file:`group_vars`
```
```
:file:`kube-node.yml`
```

wabernat replied:
done.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +162,55 @@ first run)::
+Run the following command to deploy the cluster::
 
-Grab a coffee and wait for deployment to end.
+  (metal-k8s) $ ansible-playbook -i inventory/quickstart-cluster/hosts
```
Dropped `-b` and `playbooks/deploy.yml` which are required.

wabernat replied:
I'll blame that on a miscommunication on this side of the Atlantic. 

Fixed.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,23 +226,36 @@ Similarly, we can list all deployed Helm_ applications::
+   a machine from which you didn't install MetalK8s, copy the credentials in
+   admin.conf and export its path with::
+
+       $ export KUBECONFIG=[...]/metalk8s/inventory/quickstart-cluster/artifacts/admin.conf
```
If you copy the file from some remote node, there's no need to create this elaborate directory structure, so could be
```
export KUBECONFIG=`pwd`/admin.conf
```
or similar.

wabernat replied:
done.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,23 +226,36 @@ Similarly, we can list all deployed Helm_ applications::
+
+       $ export KUBECONFIG=[...]/metalk8s/inventory/quickstart-cluster/artifacts/admin.conf
+
+2. Open port 6443 on at least one node in your cluster for remote access to
```
This is not any node, but the first `kube-master` node specifically.

wabernat replied:
fixed.

NicolasT commented:
@wabernatScality You broke this branch :cry: You somehow merged `development/1.1` into it, whilst the target is `development/1.0` and we don't want all of `development/1.1` to be merged into `development/1.0`, of course...

NicolasT commented:
@wabernatScality I'm not sure what happened, but in 8a0224a7efc3b579725689343ac609fcb652ea5b you merged in 56be6a8b6b16eac6fa25c4fb6b3568eaef32aece which was based on `development/1.1`, whilst the target for this PR is `development/1.0`. This would, in turn, cause `development/1.1` (or a subset of it, post-`development/1.0`) to be merged in `development/1.0`, which is of course not the intent.

This must be fixed...

wabernat committed:
docs: Integrate MetalK8s section from Zenko installation guide

See: https://github.com/scality/metalk8s/pull/328
Authored-by: William Abernathy <william.abernathy@scality.com>
Authored-by: Lucie Leonard <lucie.leonard@scality.com>
Signed-off-by: Nicolas Trangez <nicolas.trangez@scality.com>

wabernat commented:
We're working on it.

On Wed, Sep 12, 2018 at 10:28 AM, Nicolas Trangez <notifications@github.com>
wrote:

> @wabernatScality <https://github.com/wabernatScality> I'm not sure what
> happened, but in 8a0224a
> <https://github.com/scality/metalk8s/commit/8a0224a7efc3b579725689343ac609fcb652ea5b>
> you merged in 56be6a8
> <https://github.com/scality/metalk8s/commit/56be6a8b6b16eac6fa25c4fb6b3568eaef32aece>
> which was based on development/1.1, whilst the target for this PR is
> development/1.0. This would, in turn, cause development/1.1 (or a subset
> of it, post-development/1.0) to be merged in development/1.0, which is of
> course not the intent.
>
> This must be fixed...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scality/metalk8s/pull/328#issuecomment-420731728>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AiJ6GIL_yoP8JoikTZJQAzyFxAnl1mEFks5uaURTgaJpZM4WbVJe>
> .
>



-- 


William Abernathy

Sr. Tech Writer

San Francisco

-- 
 
<https://www.scality.com/download-idc-marketscape/?utm_source=ScalityEmail&utm_medium=Email&utm_campaign=2018Predictions>

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,101 @@
+requirements and describes sizing, configuration, and deployment. With respect
+to installation procedures, the only significant difference between a test
+cluster and a full production environment is the amount of resources required
+to implement the design.
```
Given this, should we call it a 'test' cluster in the first sentence?

lucieleonard replied:
meaning you'd prefer not calling it a 'test' cluster?

NicolasT replied:
Indeed. If this is the installation guide, I'd assume it's meant to install clusters, for testing purposes or other.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,101 @@
+| System Storage  | 16 GB  | 128 GB | 64 GB |
++-----------------+--------+--------+-------+
+
+The root file system requires at least 20 GB.
```
If the root FS should be 20GB, what is the *System Storage* referred to above? Also, why would a *Master* node require much more storage than a *Node*?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +164,55 @@ first run)::
-export this location in the shell such that the :program:`kubectl` and
-:program:`helm` tools know how to contact the cluster *kube-master* nodes, and
-authenticate properly::
+    (metal-k8s) $ export KUBECONFIG=`pwd`/admin.conf
```
This is not where `admin.conf` is located, it's in the inventory under `artifacts`.

lucieleonard replied:
Can you give me the inventory path?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -159,49 +164,55 @@ first run)::
+    (metal-k8s) $ export KUBECONFIG=`pwd`/admin.conf
 
-    (metal-k8s) $ export KUBECONFIG=`pwd`/inventory/quickstart-cluster/artifacts/admin.conf
+If your system can reach port 6443 on a master node, you can
```
Not *a* master node, but the 'first' master node (where 'first' can in theory be any of them). Basically, the master node referred to in `admin.conf`.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,23 +228,36 @@ Similarly, we can list all deployed Helm_ applications::
+1. The credentials in :file:`inventory/quickstart-cluster/artifacts/admin.conf`
+   must be present. If you are accessing the cluster using the same machine
+   from which you installed MetalK8s, this path is already set in the
+   ``make shell`` virtual environment. If you are accessing the cluster using
```
It's only set if you set it before (earlier step in these instructions).

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,23 +228,36 @@ Similarly, we can list all deployed Helm_ applications::
+   a machine from which you didn't install MetalK8s, copy the credentials in
+   admin.conf and export its path with::
+
+       $ export KUBECONFIG=[...]/metalk8s/inventory/quickstart-cluster/artifacts/admin.conf
```
I'd replace this instance with a more generic path. When copying `admin.conf` to another machine, there's no need to recreate the whole inventory directory structure.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -217,23 +228,36 @@ Similarly, we can list all deployed Helm_ applications::
+
+       $ export KUBECONFIG=[...]/metalk8s/inventory/quickstart-cluster/artifacts/admin.conf
+
+2. Open port 6443 on your cluster’s master node for remote access to cluster
```
There are multiple master nodes.

lucieleonard committed:
tox: Ignore long lines in `quickstart.rst`on Windows

Signed-off-by: Nicolas Trangez <nicolas.trangez@scality.com>

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,92 @@
-configuration.
+General Cluster Requirements
+----------------------------
+Setting up a MetalK8s test cluster quickly requires three machines
```
Same `test cluster` comment.

NicolasT replied:
'at least three machines'?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,92 @@
+
+Get the MetalK8s Git Repo
+-------------------------
+Go to the MetalK8s releases_ page and download the source code zip file
```
Then the title of this section is no longer correct  :)

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,92 @@
+Get the MetalK8s Git Repo
+-------------------------
+Go to the MetalK8s releases_ page and download the source code zip file
+for the release you wish to get.
```
'wish to get' -> 'wish to install'?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,92 @@
+To create an inventory:
+
+1. Log in to the machine to which you cloned or put the git repo, as described
+   in the previous section.
```
The Git repository is not cloned in the previous section (anymore).

lucieleonard replied:
Can we talk about a "MetalK8s project"? or "release"?

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -164,49 +160,56 @@ first run)::
+Run the following command to deploy the cluster::
 
-    (metal-k8s) $ ansible-playbook -i inventory/quickstart-cluster -b playbooks/deploy.yml
+  (metal-k8s) $ ansible-playbook -i inventory/quickstart-cluster -b playbooks/deploy.yml
```
Better use `inventory/quickstart-cluster/hosts` as the `-i` argument.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -222,23 +225,31 @@ Similarly, we can list all deployed Helm_ applications::
+2. Open port 6443 on your cluster’s master nodes for remote access to cluster
+   services.
+
+3. Run ``kubectl proxy`` from your local machine. This opens a tunnel to the
```
'Inside a `make shell` environment, run ...`

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -247,3 +258,5 @@ services and their configuration.
 .. _Grafana: https://grafana.com
 .. _Cerebro: https://github.com/lmenezes/cerebro
 .. _Kibana: https://www.elastic.co/products/kibana/
+.. _storage: (Link needed)
```
This is a TODO?

lucieleonard replied:
I need to find the way to insert a cross-reference to another file. Currently on it

NicolasT replied:
Looks like you figured it out by adding an anchor to the top of the file, and then use the `ref` role (http://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-ref) which is likely the 'best' solution.

An alternative is/was to use the `doc` role: http://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-doc

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -247,3 +258,5 @@ services and their configuration.
 .. _Cerebro: https://github.com/lmenezes/cerebro
 .. _Kibana: https://www.elastic.co/products/kibana/
+.. _storage: (Link needed)
+.. _releases: https://github.com/scality/metalk8s/releases
```
I tend to put references closed to where they're used rather than at the bottom of the file, far from their use site. This eases reading/reviewing the ReST source.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,93 @@
+
+ http_proxy=http://user;pass@<HTTP proxy IP address>:<port>
+ https_proxy=http://user;pass@<HTTPS proxy IP address>:<port>
+ no_proxy=localhost,127.0.0.1,10.*
```
This is likely incorrect, but OK for now until we are sure how to do it 'properly'.

Also, we may want to let MetalK8s handle this in 1.1.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -5,33 +5,93 @@
+ https_proxy=http://user;pass@<HTTPS proxy IP address>:<port>
+ no_proxy=localhost,127.0.0.1,10.*
+
+Download the MetalK8s Git Source
```
Not `Git`.

NicolasT commented on docs/usage/quickstart.rst:
```diff
@@ -56,105 +116,42 @@ listing all the hosts, aptly called :file:`hosts`:
-:program:`make` target which installs these in a local environment. To enter
-this environment, run :command:`make shell` (this takes a couple of seconds on
-first run)::
+Enter the MetalK8s Virtual Environment Shell
```
At no point before did you unpack the source zipfile I believe. Should this be added?

lucieleonard replied:
I'm not sure this should needs to be written, but in case... Adding a sentence in "Download the MetalK8s Source" :-)

NicolasT commented:
@lucieleonard Some last comments. Once those are in, I can do some Git magic to fixup the whole load of commits, and then we'll need approval (maybe from @wabernatScality depending on when this is done and timezone), and then it can get merged.

NicolasT commented:
@lucieleonard Oh, doc building failed in CI:

```
docs/usage/quickstart.rst:239: D002 Trailing whitespace
```

Make sure `tox -e docs` works on your machine before committing.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1002` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1002](https://github.com/scality/metalk8s/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1002) will be merged into `development/1.1` (pull request #377)





*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @ssalaues

* @alxf

* @ballot-scality

* @Zempashi

* @NicolasT

* @anurag4DSB

* @thomasdanan

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`

* :heavy_check_mark: `development/1.1`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`

* :heavy_check_mark: `development/1.1`




Please check the status of the associated issue None.

Goodbye NicolasT.

Zempashi commented on docs/usage/quickstart.rst:
```diff
@@ -56,105 +117,42 @@ listing all the hosts, aptly called :file:`hosts`:
-- Create and register the default MetalK8s 0.1 LVs and *PersistentVolumes*:
-
-  .. code-block:: yaml
+    metalk8s_lvm_drives_vg_metalk8s: ['/dev/vdb']
```
@lucieleonard this one is fine (but in quickstart), If you create an extensive documentation for storage: this is among the first piece of info to give

Zempashi commented on docs/usage/quickstart.rst:
```diff
@@ -56,105 +117,42 @@ listing all the hosts, aptly called :file:`hosts`:
-  .. code-block:: yaml
+    metalk8s_lvm_drives_vg_metalk8s: ['/dev/vdb']
 
-      metalk8s_lvm_lvs_kubevg:
```
@lucieleonard: this one is for adding storage some "partitions" to the storage. The variable name should be `metalk8s_lvm_lvs_vg_metalk8s`

Zempashi commented on docs/usage/quickstart.rst:
```diff
@@ -56,105 +117,42 @@ listing all the hosts, aptly called :file:`hosts`:
 
-      metalk8s_lvm_lvs_kubevg:
-        lv01:
-          size: 52G
```
Those size depends on the workload metalk8s will receive: This applies to all nodes.  If you have 5 nodes: the whole cluster will get 3 (lv01, lv02, lv03) * 5 nodes = 15 lv of 52g

lucieleonard replied:
3 lv for each node?
